uniflow-test/src/test/kotlin/io/uniflow/test/impl/AbstractSampleFlow.kt

```diff
@@ -1,47 +1,41 @@
 package io.uniflow.test.impl
 
 import io.uniflow.core.dispatcher.UniFlowDispatcher
-import io.uniflow.core.flow.*
+import io.uniflow.core.flow.ActionDispatcher
+import io.uniflow.core.flow.ActionReducer
+import io.uniflow.core.flow.DataFlow
+import io.uniflow.core.flow.DataPublisher
 import io.uniflow.core.flow.data.UIData
+import io.uniflow.core.flow.data.UIEvent
 import io.uniflow.core.flow.data.UIState
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.cancel
 import kotlinx.coroutines.channels.Channel
-import kotlin.reflect.KClass
 
-abstract class AbstractSampleFlow(defaultState: UIState) : DataFlow {
-    private val tag = this.toString()
+abstract class AbstractSampleFlow(val defaultState: UIState) : DataFlow, DataPublisher {
+    override val tag = this.toString()
     private val supervisorJob = SupervisorJob()
     private val coroutineScope = CoroutineScope(Dispatchers.Main + supervisorJob)
-    private val dataPublisher: SimpleDataPublisher = SimpleDataPublisher()
-    private val dataStore: UIDataStore = UIDataStore(dataPublisher, defaultState, tag)
-    private val reducer: ActionReducer = ActionReducer(dataStore, coroutineScope, UniFlowDispatcher.dispatcher.main(), Channel.BUFFERED, tag)
-    private val actionDispatcher: ActionDispatcher
-        get() = ActionDispatcher(coroutineScope, reducer, dataStore, this, tag)
 
-    final override fun getCurrentState() = actionDispatcher.getCurrentState()
-    final override fun <T : UIState> getCurrentStateOrNull(stateClass: KClass<T>): T? = actionDispatcher.getCurrentStateOrNull()
-    final override fun action(onAction: ActionFunction<UIState>): ActionFlow = actionDispatcher.action(onAction)
-    final override fun action(onAction: ActionFunction<UIState>, onError: ActionErrorFunction): ActionFlow = actionDispatcher.action(onAction, onError)
-    final override fun <T : UIState> actionOn(stateClass: KClass<T>, onAction: ActionFunction<T>): ActionFlow = actionDispatcher.actionOn(stateClass, onAction)
-    final override fun <T : UIState> actionOn(stateClass: KClass<T>, onAction: ActionFunction<T>, onError: ActionErrorFunction): ActionFlow = actionDispatcher.actionOn(stateClass, onAction, onError)
+    internal val defaultDataPublisher = simpleListPublisher(defaultState, "main")
```

Comment:
You can apply the same changes here. From -> https://github.com/uniflow-kt/uniflow-kt/pull/56/files#diff-20ab9838c64e7a207e75b2d7617f4c5e336e6c81dd4233cf6d6126fde25eef15